### PR TITLE
Add extensions-sdk watch build option

### PR DESCRIPTION
**Pull request by moeriki**
I added this to my fork because it's very convenient.

CleanShot 2021-09-06 at 15 23 27

### CODE REVIEW

**Comment by Moeriki** [comment]
Small note. The package seems to have been renamed from `extension-sdk` to `extensions-sdk` in rc92 but I couldn't find this in the release notes.

**Comment by Moeriki** [comment]
Fixed an issue with swallowed build errors.

CleanShot 2021-09-06 at 16 44 40@2x

I ended up lifting quite a bit of code from the Rollup CLI error handler.

https://github.com/rollup/rollup/blob/181f379903988819c79ed6aac7fb5af2fa35877f/cli/logging.ts#L14

Please let me know what you think.

**Comment by nickrum** [comment]
Thanks for working on this. Seems like a very usefull addition 👍

We should simplify the error message and re-use the logic for the regular build mode.

There is currently a weird issue where rollup won't rebuild the extensions if there was a syntax error in a `.vue` file, that has been fixed without doing any other changes. Not sure if this is something on our end or an issue with `rollup-plugin-vue`.

**Comment by Moeriki** [comment]
I reused the error handler for the build and simplified the error handling just a bit.

Besides the `error.url` which I don't know what it is, I find all the information logged very useful. I just deconstructed and simplified the code.

File: packages/extensions-sdk/src/cli/commands/build.ts [hunk 1]
```diff
@@ -75,13 +82,47 @@ export default async function build(options: BuildOptions): Promise<void> {
...
+		watcher.on('event', async (event) => {
+			switch (event.code) {
+				case 'ERROR': {
+					spinner.fail(chalk.bold('Failed'));
+					handleRollupError(event.error);
+					spinner.start(chalk.bold('Watching files for changes...'));
+					break;
+				}
+				case 'BUNDLE_END':
+					await event.result.write(rollupOutputOptions);
```
Comment (nickrum) [inline packages/extensions-sdk/src/cli/commands/build.ts:100]: This line can be removed as `event.result.write()` is only meant for generating additional bundles.

File: packages/extensions-sdk/src/cli/commands/build.ts [hunk 2]
```diff
@@ -75,13 +82,47 @@ export default async function build(options: BuildOptions): Promise<void> {
...
+					await event.result.close();
+					spinner.succeed(chalk.bold('Done'));
+					spinner.start(chalk.bold('Watching files for changes...'));
+					break;
+				case 'START':
+					spinner.text = 'Building Directus extension...';
+					break;
+				case 'END':
```
Comment (nickrum) [inline packages/extensions-sdk/src/cli/commands/build.ts:108]: This stray switch case can be removed.

File: packages/extensions-sdk/src/cli/commands/build.ts [hunk 3]
```diff
@@ -75,13 +82,47 @@ export default async function build(options: BuildOptions): Promise<void> {
...
+					spinner.start(chalk.bold('Watching files for changes...'));
+					break;
+				}
+				case 'BUNDLE_END':
+					await event.result.write(rollupOutputOptions);
+					await event.result.close();
+					spinner.succeed(chalk.bold('Done'));
+					spinner.start(chalk.bold('Watching files for changes...'));
+					break;
+				case 'START':
```
Comment (nickrum) [inline packages/extensions-sdk/src/cli/commands/build.ts:105]: Let's use `BUNDLE_START` here for consistency.

File: packages/extensions-sdk/src/cli/commands/build.ts [hunk 4]
```diff
@@ -100,7 +141,7 @@ function getRollupOptions(
 				styles(),
 				...plugins,
 				nodeResolve({ browser: true }),
-				commonjs({ esmExternals: true, sourceMap: false }),
+				commonjs({ esmExternals: true, sourceMap: true }),
```
Comment (nickrum) [inline packages/extensions-sdk/src/cli/commands/build.ts:144]: Curious to hear what advantages there are of enabling source map generation by the commonjs plugin.

**Review by nickrum — CHANGES_REQUESTED** [review]

File: packages/extensions-sdk/src/cli/commands/build.ts (hunk 4 above)
Comment (Moeriki) [reply packages/extensions-sdk/src/cli/commands/build.ts:144]: Oops that's committed by mistake.

I was looking to get sourcemaps of my modules to get a stacktrace when running a module within Directus. Without success.

**Review by nickrum — APPROVED** [review]

**Pull request merged**